core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -129,30 +162,35 @@ class RequestSendThread(val controllerId: Int,
           // if a broker goes down for a long time, then at some point the controller's zookeeper listener will trigger a
           // removeBroker which will invoke shutdown() on this thread. At that point, we will stop retrying.
           try {
-            channel.send(request)
-            receive = channel.receive()
-            isSendSuccessful = true
+            if (!brokerReady()) {
```

Comment:
I did this here instead of changing `blockingSendAndReceive` to make it easier to maintain the previous behaviour of logging once a connection is established.